test/include/kernels/MatDiffusionTest.h

```diff
@@ -7,22 +7,22 @@
 //* Licensed under LGPL 2.1, please see LICENSE for details
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
-#ifndef MATDIFFUSION_H
```

Comment:
Rather than re-naming this class, could we just have the MOOSE tests use MatDiffusion.C that you are moving over from PhaseField? It looks like the only missing feature there is the ability to use the old/older states.  You could just put that in the PhaseField version.  Also, you need base_name in this class as well for it to work with XFEM, right?

Reply:
I talked about this with @dschwen, and he prefers this the way you have it (keeping the old/older state option out of MatDiffusion and re-naming the testing object to MatDiffusionTest).  I'm OK either way, so I guess I'd say just keep this how you have it.

We still need that `base_name` parameter in MatDiffusion, so adding that is the only thing that you would need to change aside from sorting out the documentation build issues.

Reply:
I just added base_name option to the base class. Could you review it?

Other comments have been addressed.